services/example-website/server/templates/components/content-body.marko

```diff
@@ -31,7 +36,48 @@ $ const useIdxDownload = defaultValue(get(content, "gating.surveyType") === "idx
   <if(input.beforeBody)>
     <${input.beforeBody} context=context />
   </if>
-  <if(cm.shouldTruncate(contentMeterState))>
+  <if(idxFormId)>
+    <if(displayIdxForm)>
+      $ const body = getContentPreview({ body: content.body, selector: "p:lt(3)" });
+      <marko-web-content-body block-name=blockName obj={ body } />
+      <div class="content-page-preview-overlay" />
+      <marko-web-content-body block-name=blockName attrs={ style: "display: none;" } obj=content modifiers=["ld-json"] />
+        <marko-web-identity-x-context|{ hasUser, user, application }|>
+        <if(hasUser)>
+          <marko-web-identity-x-form-access
+            content=content
+            user=user
+            formId=idxFormId
+            application=application
+          />
+          $ console.log("authenticated", idxFormId);
+        </if>
+        <else>
+          $ console.log('!hasUser')
+          <marko-web-identity-x-non-auth-identify|{ user: identifiedUser }|>
+            <marko-web-identity-x-form-access
+              content=content
+              user=identifiedUser
+              formId=idxFormId
+              application=application
+            />
+            $ console.log("Identitfied", identifiedUser);
+          </marko-web-identity-x-non-auth-identify>
+        </else>
+      </marko-web-identity-x-context>
```

Comment:
Don't create two separate calls and a wrapping component just for a promise. Handle this logic inside the form-access component. For that matter, you don't need the identity-x-context wrapper either, just call your component directly and handle the context promise resolution inside it. The simpler it is to implement, the less the chance of it getting messed up when rolled out.